lib/capabilitySets.js

```diff
@@ -0,0 +1,204 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const bedrock = require('bedrock');
+const database = require('bedrock-mongodb');
+const {promisify} = require('util');
+const utils = require('./utils');
+const {util: {BedrockError}} = bedrock;
+
+// load config defaults
+require('./config');
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  const collections = ['profile-profileAgentCapabilitySet'];
+  await promisify(database.openCollections)(collections);
+
+  await promisify(database.createIndexes)([{
+    // cover capability set queries by profileAgent ID
+    collection: 'profile-profileAgentCapabilitySet',
+    fields: {'capabilitySet.profileAgent': 1},
+    options: {unique: true, background: false}
+  }]);
+});
+
+// create a capability set
+exports.create = async ({capabilitySet}) => {
+  assertCapabilitySet(capabilitySet);
+  if(capabilitySet.sequence !== 0) {
+    throw new BedrockError(
+      'Profile agent capability set sequence must be zero when created.',
+      'InvalidStateError', {
+        public: true,
+        httpStatusCode: 400
+      });
+  }
+
+  // insert the capability set and get the updated record
+  const now = Date.now();
+  const meta = {created: now, updated: now};
+  let record = {
+    meta,
+    capabilitySet
+  };
+  try {
+    const collection = utils.getCollection('profile-profileAgentCapabilitySet');
+    const result = await collection.insert(record, database.writeOptions);
+    record = result.ops[0];
+  } catch(e) {
+    if(!database.isDuplicateError(e)) {
+      throw e;
+    }
+    throw new BedrockError(
+      'Duplicate profile agent capability set.',
+      'DuplicateError', {
+        public: true,
+        httpStatusCode: 409
+      }, e);
+  }
+  return record;
+};
+
+// get a capability set by profile agent ID
+exports.get = async ({profileAgentId}) => {
+  assert.string(profileAgentId, 'profileAgentId');
+
+  const query = {'capabilitySet.profileAgent': profileAgentId};
+  const collection = utils.getCollection('profile-profileAgentCapabilitySet');
+  const record = await collection.findOne(query,
+    {_id: 0, capabilitySet: 1, meta: 1});
+  if(!record) {
+    const details = {
+      profileAgent: profileAgentId,
+      httpStatusCode: 404,
+      public: true
+    };
+    throw new BedrockError(
+      'Profile agent capability set not found.',
+      'NotFoundError', details);
+  }
+  return record;
+};
+
+// get all capability sets for a profile agent
+exports.getAll = async ({profileAgentId} = {}) => {
```

Comment:
There can't be more than one, so we can remove this, right?

Reply:
Removed https://github.com/digitalbazaar/bedrock-profile/pull/1/commits/7af43439e92bf96fa245758e4f385cf9aefc158d#diff-0b71788d914d266bcb6d38de72ec456fL86